### PR TITLE
refactor(scenes): automatically bind props to logic in scenes

### PR DESCRIPTION
**Pull request by mariusandra**
## Problem

Trying to make our lives easier by binding each scene's keyed logic to the scene automatically.

Each `SceneExport` has a `paramsToProps` function, which specifies how to build a logic from the URL. Despite this function existing, you had to convert the params to props a second time when getting values/actions from the scene's logic.

## Changes

- This PR automatically binds the scene's logic to the scene's component
- Moves a few props to selectors where it makes sense
- Removes some `<BindLogic>` that's no longer needed
- Updates the "how to build a scene" page
- Small TODO outside the scope here to work better with kea-forms (the forms should use the bound props as well)
- I didn't modify the cohorts page, since I know work is happening there.

## How did you test this code?

- Clicked around in each of the affected pages (dashboard, experiment, feature flag, persons, actions), saw that things still worked

### CODE REVIEW

File: frontend/src/scenes/dashboard/Dashboard.tsx [hunk 1]
```diff
@@ -23,21 +23,26 @@ interface Props {
...
     logic: dashboardLogic,
-    paramsToProps: ({ params: { id } }): DashboardLogicProps => ({ id: parseInt(id) }),
+    paramsToProps: ({ params: { id, shareToken, placement } }: { params: Props }): DashboardLogicProps => ({
+        id: id ? parseInt(id) : undefined,
+        shareToken,
+        placement,
+    }),
 }
 
 export function Dashboard({ id, shareToken, placement }: Props = {}): JSX.Element {
```
[the remark this answers is not in the document]
Comment (mariusandra) [reply frontend/src/scenes/dashboard/Dashboard.tsx:35]: I thought about it, and had actually removed it... but put it back. It felt weird that the shared dashboard "scene" needs to know this deeply about dashboard internals. This tiny wrapper and the naming around it is not that ideal either... 🤔

File: frontend/src/scenes/experiments/Experiment.tsx [hunk 2]
```diff
@@ -51,8 +50,9 @@ export const scene: SceneExport = {
     }),
 }
 
-export function Experiment_({ id }: { id?: Experiment['id'] } = {}): JSX.Element {
+export function Experiment_(): JSX.Element {
```
[the remark this answers is not in the document]
Comment (mariusandra) [reply frontend/src/scenes/experiments/Experiment.tsx:53]: I think `Experiment` was a type that was imported...

**Review by Twixes — CHANGES_REQUESTED** [review]
All good except I get this when I open the project homepage with no home dashboard selected:
localhost_8000_home

**Review by Twixes — APPROVED** [review]
Let's gooo

**Pull request merged**